Library/Homebrew/formula_installer.rb

```diff
@@ -1123,21 +1117,7 @@ def fetch
 
     return if only_deps?
 
-    if pour_bottle?(output_warning: true)
-      begin
-        downloader.fetch
-      rescue Exception => e # rubocop:disable Lint/RescueException
-        raise if Homebrew::EnvConfig.developer? ||
-                 Homebrew::EnvConfig.no_bottle_source_fallback? ||
-                 force_bottle? ||
-                 e.is_a?(Interrupt)
-
-        @pour_failed = true
-        onoe e.message
-        opoo "Bottle installation failed: building from source."
-        fetch_dependencies
-      end
-    end
+    downloader.fetch if pour_bottle?(output_warning: true)
     return if pour_bottle?
 
     formula.fetch_patches
```

Comment:
Maybe try avoid the double `pour_bottle?` check here. Can't quite make the proper suggestion diff because GitHub diff cuts off but:

```suggestion
    unless pour_bottle?(output_warning: true)
      formula.fetch_patches
      formula.resources.each(&:fetch)
    end
    downloader.fetch
  end
```